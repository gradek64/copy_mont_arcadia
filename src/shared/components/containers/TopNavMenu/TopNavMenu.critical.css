/** @define TopNavMenu */

.TopNavMenu {
  opacity: 0;
}
