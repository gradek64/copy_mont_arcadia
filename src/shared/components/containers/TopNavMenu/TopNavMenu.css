/** @define TopNavMenu */
@lost flexbox flex;

.TopNavMenu {
  lost-flex-container: column;
  position: fixed;
  top: 0;
  width: 80%; /* fallback no vw */
  width: 80vw;
  height: 100%;
  transition: transform 0.4s ease-in-out;
  transform: translateX(-100%);
  will-change: transform;
  font-size: $font-size-navigation;
  z-index: 4;
  overflow: hidden;
  pointer-events: none;

  &:focus {
    outline: none;
  }

  &.is-open {
    transform: none;
    opacity: 1;
    -webkit-overflow-scrolling: touch;
  }

  &-arrowBack {
    position: absolute;
    top: 50%;
    left: $navigation-item-padding-vertical;
    height: $navigation-item-arrow-height;
    width: $navigation-item-arrow-width;
    margin-top: calc(-0.5 * $navigation-item-arrow-height);
    background: url("public/$brand/images/left-arrow.svg") no-repeat center;
    background-size: contain;
  }

  &-home {
     position: absolute;
     left: $navigation-item-padding-vertical;
     margin-top: $navigation-home-icon-margin-top;
     height: $navigation-home-icon-height;
     width: 18px;
     background: url("public/$brand/images/home-icon.svg") no-repeat center;
     background-size: contain;
  }

  &-popCategoryButton {
    width: 100%;
  }

  &-homeButton {
    width: 100%;
  }

  &-groupHeader {
    margin: 0;
    flex-grow: 0;
    background-color: $navigation-group-header-background;
    font-size: $navigation-group-header-font-size;
    font-weight: $navigation-group-header-font-weight;
    padding: $navigation-group-header-padding-vertical $navigation-item-padding-horizontal;
    color: $navigation-group-header-font-color;
    border-top: 1px solid $navigation-group-header-border-color;
    border-bottom: 1px solid $navigation-group-header-border-color;
    display: $navigation-group-header-visibility;
  }

  &-parentListBlock {
    position: relative;
    height: 100%;
    margin: 0;
    padding-left: 0;
    overflow-y: scroll;
    background: $white;
    border-right: solid 1px $navigation-border-right-color;
    pointer-events: auto;

    & > :last-child {
      border-bottom: none;
    }
  }

  &-userLocatorContainer {
    padding: 10px $navigation-item-padding-horizontal;
    background-color: $white;
  }

  &-shopByCategory {
    background: $navigation-category-section-background;
  }
  &-userDetails {
    background: $navigation-details-section-background;
  }
  &-helpAndInfo {
    background: $navigation-help-section-background;
  }
  &-close {
    color: $navigation-item-font-color;
    font-weight: 300;
    text-decoration: underline;
    width: 100%;
    text-align: right;
    box-sizing: border-box;
    padding: $navigation-item-padding-vertical $navigation-item-padding-horizontal;
  }
}

/* postcss-bem-linter: ignore */
.TopNavMenu.is-open .TopNavMenu-parentListBlock {
  box-shadow: $navigation-box-shadow;
}
