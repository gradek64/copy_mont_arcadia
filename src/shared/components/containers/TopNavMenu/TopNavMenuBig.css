/** @define TopNavMenuBig */

.TopNavMenuBig {
  display: none;
  background-color: $topnavmenu-big-background-color;
  color: $topnavmenu-big-font-color;
  width: 100%;

  &-categories {
    display: flex;
    align-items: center;
    justify-content: $topnavmenu-big-justify-content;
    max-width: $laptop-end-width;
    margin: 0 auto;
  }

  &-category {
    text-transform: uppercase;
    text-align: center;
    font-size: $topnavmenu-big-font-size;
    padding: $topnavmenu-big-padding;
  }
}

@media $min-tablet {
  .TopNavMenuBig {
    display: block;
  }
}
