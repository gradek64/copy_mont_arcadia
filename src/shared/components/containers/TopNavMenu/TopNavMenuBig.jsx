import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

@connect((state) => ({
  categories: state.navigation.productCategories,
}))
class TopNavMenuBig extends Component {
  static propTypes = {
    categories: PropTypes.array,
    className: PropTypes.string,
  }

  render() {
    const { categories, className } = this.props
    return (
      <div className={`TopNavMenuBig ${className || ''}`}>
        <div className="TopNavMenuBig-categories">
          {categories.map((cat) => {
            return (
              <div key={cat.categoryId} className="TopNavMenuBig-category">
                {cat.label}
              </div>
            )
          })}
        </div>
      </div>
    )
  }
}

export default TopNavMenuBig
