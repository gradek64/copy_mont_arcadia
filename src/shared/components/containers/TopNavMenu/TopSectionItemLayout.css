/** @define TopSectionItemLayout; weak */

.TopSectionItemLayout {
    position: relative;
    padding-right: 20px;

  &-leftIconWrapper {
    position:absolute;
    width: 30px;
    text-align: center;
  }
  &-textWrapper {
    position: relative;
    left: 35px;
    display: inline-block;
  }
  &-arrow{
    position: absolute;
    top: 50%;
    right: $navigation-item-padding-horizontal;
    height: $navigation-item-arrow-height;
    width: $navigation-item-arrow-width;
    margin-top: calc(-0.5 * $navigation-item-arrow-height);
    background: url("public/$brand/images/nav-arrow.svg") no-repeat center;
    background-size: contain;
    
    &--downArrow, &--upArrow {
      transform: rotate(90deg);
      right: calc($navigation-item-padding-horizontal + 3);
    }
    
    &--upArrow{
      transform: rotate(-90deg);
    }
  }
}                                                          