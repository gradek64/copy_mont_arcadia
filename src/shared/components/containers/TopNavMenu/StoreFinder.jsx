import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Image from '../../common/Image/Image'
import Form from '../../common/FormComponents/Form/Form'
import storeLocatorConsts from '../../../constants/storeLocator'
import UserLocatorInput from '../../common/UserLocatorInput/UserLocatorInput'
import { selectCountry } from '../../../actions/components/StoreLocatorActions'
import {
  searchStores,
  setSelectedPlace,
  resetPredictions,
  resetSearchTerm,
  resetSelectedPlace,
} from '../../../actions/components/UserLocatorActions'
import { setFormField } from '../../../actions/common/formActions'
import TopSectionItemLayout from './TopSectionItemLayout'
import { isFeatureYextEnabled } from '../../../../shared/selectors/featureSelectors'
import { isUsingCurrentLocation } from '../../../../shared/selectors/formsSelectors'
import { getBrandName } from '../../../../shared/selectors/configSelectors'

// selectors
import { getShippingDestination } from '../../../selectors/shippingDestinationSelectors'
import {
  getGeoLocatorCoordinatesLong,
  getGeoLocatorCoordinatesLat,
  getPredictionDescription,
} from '../../../selectors/userLocatorSelectors'

@connect(
  (state) => ({
    selectedCountry: getShippingDestination(state), // assigned the value returned by getShippingDestination as this components now work in tandem
    isFeatureYextEnabled: isFeatureYextEnabled(state),
    isUsingCurrentLocation: isUsingCurrentLocation(state),
    brandName: getBrandName(state),
    userLat: getGeoLocatorCoordinatesLat(state),
    userLong: getGeoLocatorCoordinatesLong(state),
    getPrediction: getPredictionDescription(state),
  }),
  {
    searchStores,
    setSelectedPlace,
    resetPredictions,
    resetSearchTerm,
    resetSelectedPlace,
    setFormField,
    selectCountry,
  }
)
class StoreFinder extends Component {
  state = {
    showUserLocatorInput: false,
  }

  static propTypes = {
    selectedCountry: PropTypes.string,
    storeLocatorLandingPage: PropTypes.bool,
    searchStores: PropTypes.func,
    resetPredictions: PropTypes.func,
    resetSearchTerm: PropTypes.func,
    resetSelectedPlace: PropTypes.func,
    setFormField: PropTypes.func,
    isFeatureYextEnabled: PropTypes.bool,
    selectCountry: PropTypes.func,
    getPrediction: PropTypes.string,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  UNSAFE_componentWillReceiveProps({ selectedCountry }) {
    if (selectedCountry && selectedCountry !== this.props.selectedCountry) {
      this.searchStoresByCountry(selectedCountry)
      if (
        this.state.showUserLocatorInput &&
        selectedCountry !== storeLocatorConsts.defaultCountry
      ) {
        this.setState({
          showUserLocatorInput: false,
        })
      }
    }
  }

  clearSearchField = () => {
    const {
      setFormField,
      resetPredictions,
      resetSearchTerm,
      resetSelectedPlace,
    } = this.props
    setFormField('userLocator', 'userLocation', '')
    resetPredictions()
    resetSearchTerm()
    resetSelectedPlace()
  }

  submitHandler = (e) => {
    if (e) e.preventDefault()
    /*  setTimeout is used here to move the below code to the end of event queue.
      * This is because the prediction is updated in state, however not quick enough for 
      * the getPrediction selector below to pick up the change.
    */
    setTimeout(() => {
      const {
        isFeatureYextEnabled,
        brandName,
        isUsingCurrentLocation,
        selectedCountry,
        userLat,
        userLong,
        searchStores,
        getPrediction,
      } = this.props
      if (
        isFeatureYextEnabled &&
        selectedCountry === storeLocatorConsts.defaultCountry
      ) {
        if (isUsingCurrentLocation) {
          const url = `${
            storeLocatorConsts.yextProdLinks[brandName]
          }?q=${userLat},${userLong}&qp=My%20Location&l=en_GB`
          window.location = url
        } else {
          const url = `${
            storeLocatorConsts.yextProdLinks[brandName]
          }?q=${getPrediction}`
          window.location = url
        }
      } else {
        searchStores()
      }
    })
  }

  searchStoresByCountry = (country) => {
    const {
      storeLocatorLandingPage,
      searchStores,
      selectCountry,
      isFeatureYextEnabled,
    } = this.props
    if (!isFeatureYextEnabled) {
      selectCountry(country)
    }
    if (
      storeLocatorLandingPage &&
      country !== storeLocatorConsts.defaultCountry
    ) {
      // Avoiding the scenario where the User selects UK and a place in the UK,
      // then he selects US and then again UK seeing the search field already
      // filled with the previous data.
      this.clearSearchField()
      searchStores()
    }
  }

  get showInput() {
    const { selectedCountry } = this.props
    return selectedCountry === storeLocatorConsts.defaultCountry
  }

  handleRowClick = () => {
    if (!this.showInput) {
      this.submitHandler()
    } else {
      this.toggleUserLocatorInput()
    }
  }

  toggleUserLocatorInput() {
    this.setState({
      showUserLocatorInput: !this.state.showUserLocatorInput,
    })
  }

  doesBrandSupportYext(brandName) {
    const unsupportedBrands = ['wallis', 'evans']
    return !unsupportedBrands.includes(brandName)
  }

  render() {
    const { l } = this.context
    const { showUserLocatorInput } = this.state
    const { selectedCountry, isFeatureYextEnabled, brandName } = this.props

    const categoryArrow = <span className="Categories-arrow" />

    const showHideArrow = (
      <span
        role="presentation"
        className={`TopSectionItemLayout-arrow TopSectionItemLayout-arrow--${
          showUserLocatorInput ? 'up' : 'down'
        }Arrow`}
      />
    )

    const rightIcon =
      selectedCountry === storeLocatorConsts.defaultCountry
        ? showHideArrow
        : categoryArrow

    return (
      <Form
        id="store-finder-form"
        className="StoreFinder"
        onSubmit={this.submitHandler}
      >
        <div
          className="rowClick"
          role="presentation"
          onClick={this.handleRowClick}
        >
          <TopSectionItemLayout
            leftIcon={
              <Image
                className="StoreFinder-mapMarker"
                alt="Store locator map marker"
                src={'/assets/{brandName}/images/map-marker.svg'}
              />
            }
            text={'Store Finder'}
            rightIcon={rightIcon}
            expanded={showUserLocatorInput}
          />
        </div>

        {showUserLocatorInput && (
          <div className="StoreFinder-userLocatorInput">
            <UserLocatorInput
              selectedCountry={selectedCountry}
              submitHandler={this.submitHandler}
            />
            {isFeatureYextEnabled &&
              this.doesBrandSupportYext(brandName) && (
                <div className="StoreFinder-findInterStoresLink">
                  <a href="/store-locator">{l`Find International Stores`}</a>
                </div>
              )}
          </div>
        )}
      </Form>
    )
  }
}

export default StoreFinder
