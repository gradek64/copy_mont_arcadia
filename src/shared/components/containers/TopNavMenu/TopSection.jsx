import React from 'react'
// components
import StoreFinder from './StoreFinder'
import ShippingDestinationMobile from '../../common/ShippingDestination/ShippingDestinationMobile'

const TopSection = () => (
  <div>
    <StoreFinder locator="TopNavMenu" />
    <ShippingDestinationMobile />
  </div>
)

export default TopSection
