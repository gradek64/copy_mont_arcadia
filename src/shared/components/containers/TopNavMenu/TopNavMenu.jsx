import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { browserHistory } from 'react-router'
import * as topNavMenuActions from '../../../actions/components/TopNavMenuActions'
import * as navigationActions from '../../../actions/common/navigationActions'
import * as authActions from '../../../actions/common/authActions'
import ListItemLink from '../../../components/common/ListItemLink/ListItemLink'
import Categories from '../../../components/common/Categories/Categories'
import SandBox from '../../../components/containers/SandBox/SandBox'
import TopSection from './TopSection'
import cmsConsts from '../../../constants/cmsConsts'
import espots from '../../../constants/espotsMobile'
import KEYS from '../../../constants/keyboardKeys'
import navigationConsts from '../../../../shared/constants/navigation'
import { analyticsGlobalNavClickEvent } from '../../../analytics/tracking/site-interactions'
import { isFeatureDeferCmsContentEnabled } from '../../../selectors/featureSelectors'

@connect(
  (state) => ({
    topNavMenuOpen: state.topNavMenu.open,
    authentication: state.auth.authentication,
    categoryHistory: state.navigation.categoryHistory,
    mustScrollToTop: state.topNavMenu.scrollToTop,
    isFeatureDeferCmsContentEnabled: isFeatureDeferCmsContentEnabled(state),
  }),
  { ...topNavMenuActions, ...authActions, ...navigationActions }
)
class TopNavMenu extends Component {
  static propTypes = {
    mustScrollToTop: PropTypes.bool,
    topNavMenuOpen: PropTypes.bool,
    toggleTopNavMenu: PropTypes.func,
    authentication: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    categoryHistory: PropTypes.array,
    popCategoryHistory: PropTypes.func,
    toggleScrollToTop: PropTypes.func,
    closeTopNavMenu: PropTypes.func,
    isFeatureDeferCmsContentEnabled: PropTypes.bool.isRequired,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  componentDidMount() {
    document.addEventListener('keydown', this.onKeydown)
  }

  componentDidUpdate() {
    const { mustScrollToTop, toggleScrollToTop } = this.props
    if (mustScrollToTop) {
      this.scrollToTop()
      toggleScrollToTop()
    }
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.onKeydown)
  }

  onRef = (element) => {
    if (element) {
      this.element = element
    }
  }

  onKeydown = (event) => {
    if (event.keyCode === KEYS.ESCAPE) {
      this.props.closeTopNavMenu()
    }
  }

  onHomeClick = () => {
    const { toggleTopNavMenu } = this.props
    browserHistory.push('/')
    toggleTopNavMenu()
    analyticsGlobalNavClickEvent('home')
  }

  popCategoryHistory = (e) => {
    const { popCategoryHistory } = this.props
    e.preventDefault()
    popCategoryHistory()
  }

  logout = () => {
    const { toggleTopNavMenu } = this.props
    toggleTopNavMenu()
    analyticsGlobalNavClickEvent('logout')
  }

  scrollToTop = () => {
    document.getElementsByClassName(
      'TopNavMenu-parentListBlock'
    )[0].scrollTop = 0
  }

  render() {
    const {
      authentication,
      categoryHistory,
      topNavMenuOpen,
      isFeatureDeferCmsContentEnabled,
    } = this.props
    const { l } = this.context
    const currentCat = categoryHistory[categoryHistory.length - 1]
    const currentCategoryName = currentCat ? currentCat.label : ''
    const linkToPreviousCategoryClass = currentCat
      ? 'ListItemLink ListItemLink-backLabel is-active is-centered'
      : 'ListItemLink'

    return (
      <div
        className={`TopNavMenu${topNavMenuOpen ? ' is-open' : ''} hideinapp`}
        tabIndex="0"
        ref={this.onRef}
      >
        <div
          key={`top-nav-${authentication || 'default'}`}
          className="TopNavMenu-parentListBlock"
        >
          <TopSection />
          <div className="TopNavMenu-shopByCategory">
            <div className="TopNavMenu-groupHeader">
              {l(navigationConsts.SHOP_BY_CATEGORY_GROUP_LABEL)}
            </div>
            <button
              className="TopNavMenu-homeButton"
              onClick={this.onHomeClick}
            >
              <ListItemLink className="ListItemLink ListItemLink-homeLink is-active">
                <span className="TopNavMenu-home" />
                {l`Home`}
              </ListItemLink>
            </button>
            {currentCategoryName ? (
              <button
                className="TopNavMenu-popCategoryButton"
                onClick={this.popCategoryHistory}
              >
                <ListItemLink className={linkToPreviousCategoryClass}>
                  <span className="TopNavMenu-arrowBack" />
                  <a>{currentCategoryName}</a>
                </ListItemLink>
              </button>
            ) : (
              ''
            )}
            <Categories
              type={navigationConsts.PRODUCT_CATEGORIES_GROUP_TYPE}
              currentCatId={currentCat && currentCat.categoryId}
            />
          </div>

          <div className="TopNavMenu-userDetails">
            <div className="TopNavMenu-groupHeader">
              {l(navigationConsts.YOUR_DETAILS_GROUP_LABEL)}
            </div>
            <Categories type={navigationConsts.USER_DETAILS_GROUP_TYPE} />
          </div>

          <div className="TopNavMenu-helpAndInfo">
            <div className="TopNavMenu-groupHeader">
              {l(navigationConsts.HELP_AND_INFORMATION_GROUP_LABEL)}
            </div>
            <Categories type={navigationConsts.HELP_AND_INFO_GROUP_TYPE} />
          </div>
          <div className="TopNavMenu-socialLinks">
            <div className="TopNavMenu-groupHeader">{l`Follow Us`}</div>
            <SandBox
              cmsPageName={espots.navMenu[0]}
              shouldGetContentOnFirstLoad
              isInPageContent
              contentType={cmsConsts.ESPOT_CONTENT_TYPE}
              lazyLoad={isFeatureDeferCmsContentEnabled}
            />
          </div>
          <div // eslint-disable-line jsx-a11y/no-static-element-interactions
            className="TopNavMenu-close"
            onClick={this.props.toggleTopNavMenu}
          >
            {l`Close menu`}
          </div>
        </div>
      </div>
    )
  }
}

export default TopNavMenu
