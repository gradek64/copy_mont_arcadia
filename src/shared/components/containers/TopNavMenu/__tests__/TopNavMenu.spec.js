import { browserHistory } from 'react-router'
import testComponentHelper from 'test/unit/helpers/test-component'
import KEYS from '../../../../constants/keyboardKeys'
import TopNavMenu from '../TopNavMenu'
import { analyticsGlobalNavClickEvent } from '../../../../analytics/tracking/site-interactions'

jest.mock('../../../../analytics/tracking/site-interactions', () => ({
  analyticsGlobalNavClickEvent: jest.fn(),
}))

const toggleTopNavMenu = jest.fn()
const closeTopNavMenu = jest.fn()
const toggleScrollToTop = jest.fn()

const props = {
  categoryHistory: [{ label: 'testCategory' }],
}

browserHistory.push = jest.fn()

const renderComponent = testComponentHelper(TopNavMenu.WrappedComponent)

describe('@renders', () => {
  it('renders in default state', () => {
    expect(renderComponent(props).getTree()).toMatchSnapshot()
  })
  it('renders in open state - menu is open', () => {
    const { wrapper } = renderComponent({
      ...props,
      topNavMenuOpen: true,
    })
    expect(wrapper.find('.TopNavMenu').hasClass('is-open')).toBe(true)
  })
})

describe('@lifecycle', () => {
  it('call scrollToTop on componentDidUpdate', () => {
    const dummyElement = [{ scrollTop: 5 }]
    document.getElementsByClassName = () => dummyElement

    const { instance } = renderComponent({
      ...props,
      mustScrollToTop: true,
      toggleScrollToTop,
    })
    instance.componentDidUpdate()
    expect(instance.props.toggleScrollToTop).toHaveBeenCalledTimes(1)
  })
})

describe('@functions', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('`toggleTopNavMenu()` prop called 1 time when `logout()` method is called', () => {
    const { instance } = renderComponent({
      ...props,
      topNavMenuOpen: true,
      toggleTopNavMenu,
    })

    instance.logout()

    expect(instance.props.toggleTopNavMenu).toHaveBeenCalledTimes(1)
    expect(analyticsGlobalNavClickEvent).toHaveBeenCalledWith('logout')
  })
  it('`closeTopNavMenu()` prop called 1 time when `onKeydown(...)` method is called', () => {
    const { instance } = renderComponent({
      ...props,
      topNavMenuOpen: true,
      closeTopNavMenu,
    })

    instance.onKeydown({
      keyCode: KEYS.ESCAPE,
    })

    expect(closeTopNavMenu).toHaveBeenCalledTimes(1)
  })
  it('`onHomeClick()` calls the `toggleTopNavMenu` prop and the `analyticsGlobalNavClickEvent` with the correct arguments', () => {
    const { instance } = renderComponent({
      ...props,
      topNavMenuOpen: true,
      toggleTopNavMenu,
    })
    instance.onHomeClick()

    expect(browserHistory.push).toHaveBeenCalledWith('/')
    expect(instance.props.toggleTopNavMenu).toHaveBeenCalled()
    expect(analyticsGlobalNavClickEvent).toHaveBeenCalledWith('home')
  })
  it('`popCategoryHistory()` calls the `popCategoryHistory` prop and the `analyticsGlobalNavClickEvent` with the correct arguments', () => {
    const { instance } = renderComponent({
      ...props,
      topNavMenuOpen: true,
      popCategoryHistory: jest.fn(),
    })
    instance.popCategoryHistory({ preventDefault: jest.fn() })

    expect(instance.props.popCategoryHistory).toHaveBeenCalled()
  })
})
