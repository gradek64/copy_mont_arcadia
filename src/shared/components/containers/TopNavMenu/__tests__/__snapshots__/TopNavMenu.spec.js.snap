// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@renders renders in default state 1`] = `
<div
  className="TopNavMenu hideinapp"
  tabIndex="0"
>
  <div
    className="TopNavMenu-parentListBlock"
    key="top-nav-default"
  >
    <TopSection />
    <div
      className="TopNavMenu-shopByCategory"
    >
      <div
        className="TopNavMenu-groupHeader"
      >
        Shop By Category
      </div>
      <button
        className="TopNavMenu-homeButton"
        onClick={[Function]}
      >
        <ListItemLink
          className="ListItemLink ListItemLink-homeLink is-active"
        >
          <span
            className="TopNavMenu-home"
          />
          Home
        </ListItemLink>
      </button>
      <button
        className="TopNavMenu-popCategoryButton"
        onClick={[Function]}
      >
        <ListItemLink
          className="ListItemLink ListItemLink-backLabel is-active is-centered"
        >
          <span
            className="TopNavMenu-arrowBack"
          />
          <a>
            testCategory
          </a>
        </ListItemLink>
      </button>
      <Connect(Categories)
        type="productCategories"
      />
    </div>
    <div
      className="TopNavMenu-userDetails"
    >
      <div
        className="TopNavMenu-groupHeader"
      >
        Your Details
      </div>
      <Connect(Categories)
        type="userDetailsGroup"
      />
    </div>
    <div
      className="TopNavMenu-helpAndInfo"
    >
      <div
        className="TopNavMenu-groupHeader"
      >
        Help & Information
      </div>
      <Connect(Categories)
        type="helpAndInfoGroup"
      />
    </div>
    <div
      className="TopNavMenu-socialLinks"
    >
      <div
        className="TopNavMenu-groupHeader"
      >
        Follow Us
      </div>
      <Connect(SandBox)
        cmsPageName="navSocial"
        contentType="espot"
        isInPageContent={true}
        shouldGetContentOnFirstLoad={true}
      />
    </div>
    <div
      className="TopNavMenu-close"
    >
      Close menu
    </div>
  </div>
</div>
`;
